R/database.R

```diff
@@ -160,24 +188,56 @@ get_ssm_by_samples = function(these_sample_ids,
       maf_path = glue::glue(maf_template)
       full_maf_path =  paste0(config::get("project_base"), maf_path)
       message(paste("using existing merge:", full_maf_path))
-      maf_df_merge = fread_maf(full_maf_path) %>%
-        dplyr::filter(Tumor_Sample_Barcode %in% these_sample_ids)
-      #subset maf to only include first 43 columns (default)
-      if(basic_columns){maf_df_merge = dplyr::select(maf_df_merge, c(1:45))}
-      #subset maf to a specific set of columns (defined in maf_cols)
-      if(!is.null(maf_cols) && !basic_columns){maf_df_merge = dplyr::select(maf_df_merge, all_of(maf_cols))}
+      if(engine=="fread_maf"){
+        if(basic_columns){
+          maf_df_merge = fread_maf(full_maf_path,select_cols = c(1:45)) %>%
+            dplyr::filter(Tumor_Sample_Barcode %in% these_sample_ids) %>%
```

Comment:
Do you want to put this and the next line outside the ifelse check, since both filters are the same?
EDITED:: It can also probably be it's own non-exported small function, since it is also used below for the augmented variants

Reply:
The better approach is actually to have these passed to fread_maf so it can lazily drop the non-matching lines on the fly rather than storing them in memory only to have them dropped a few lines later. I think Adam can work on this as a refinement.